homeassistant/components/launch_library/sensor.py

```diff
@@ -39,39 +43,86 @@ async def async_setup_platform(
     async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Create the launch sensor."""
-    name = config[CONF_NAME]
-    session = async_get_clientsession(hass)
-    launches = PyLaunches(session)
+    """Import Launch Library configuration from yaml."""
+    _LOGGER.warning(
+        "Configuration of the launch_library platform in YAML is deprecated and will be "
+        "removed in Home Assistant 2022.4; Your existing configuration "
+        "has been imported into the UI automatically and can be safely removed "
+        "from your configuration.yaml file"
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the sensor platform."""
 
-    async_add_entities([LaunchLibrarySensor(launches, name)], True)
+    name = entry.data.get(CONF_NAME, DEFAULT_NAME)
 
+    coordinator = hass.data[DOMAIN]
 
-class LaunchLibrarySensor(SensorEntity):
-    """Representation of a launch_library Sensor."""
+    async_add_entities(
+        [
+            NextLaunchSensor(coordinator, entry.entry_id, name),
+        ]
+    )
 
-    _attr_icon = "mdi:rocket"
 
-    def __init__(self, api: PyLaunches, name: str) -> None:
-        """Initialize the sensor."""
-        self.api = api
+class NextLaunchSensor(CoordinatorEntity, SensorEntity):
+    """Representation of the next launch sensor."""
+
+    _attr_attribution = ATTRIBUTION
+    _attr_icon = "mdi:rocket-launch"
+    _next_launch: Launch | None = None
+
+    def __init__(
+        self, coordinator: DataUpdateCoordinator, entry_id: str, name: str
+    ) -> None:
+        """Initialize a Launch Library entity."""
+        super().__init__(coordinator)
         self._attr_name = name
+        self._attr_unique_id = f"{entry_id}_next_launch"
+
+    @property
+    def native_value(self) -> str | None:
+        """Return the state of the sensor."""
+        if self._next_launch is None:
+            return None
```

Comment:
```suggestion
```

This check is not needed, it's handled by the `available` property

Reply:
Same as above, it is initiated as none, thus needs to be handled